spec/support/http_method_shim.rb

```diff
@@ -3,34 +3,29 @@
 # supported in Rails 5+. Since we support back to 4, we need some sort of shim
 # to avoid super noisy deprecations when running tests.
 module RoutingHTTPMethodShim
-  def get(path, params = {}, headers = nil)
-    super(path, params: params, headers: headers)
+  def get(path, **args)
+    super(path, args[:params], args[:headers])
   end
 
-  def post(path, params = {}, headers = nil)
-    super(path, params: params, headers: headers)
+  def post(path, **args)
+    super(path, args[:params], args[:headers])
   end
 
-  def put(path, params = {}, headers = nil)
-    super(path, params: params, headers: headers)
+  def put(path, **args)
+    super(path, args[:params], args[:headers])
   end
 end
 
 module ControllerHTTPMethodShim
-  def get(path, params = {})
-    super(path, params: params)
-  end
-
-  def post(path, params = {})
-    super(path, params: params)
-  end
-
-  def put(path, params = {})
-    super(path, params: params)
+  def process(action, http_method = 'GET', **args)
+    if as = args.delete(:as)
+      @request.headers['Content-Type'] = Mime[as].to_s
+    end
+    super(action, http_method, args[:params], args[:session], args[:flash])
```

Comment:
What about headers?

Reply:
This is calling Rails 4's and in Rails 4 adding headers is not supported.